src/mimemail.erl

```diff
@@ -2441,4 +2465,70 @@ dkim_sign_test_() ->
 				   "T97QadH42PT6XmO2v01q5nhMgNE4yQyf9DBJs=">>, DkimHdrVal)
 	  end}].
 
+dkim_sign_ed25519_test_() ->
+	%% * sign using test/fixtures/dkim*.pem
+	{ok, PrivKey} = file:read_file("test/fixtures/dkim-ed25519-private.pem"),
+	[{"Sign simple",
```

Comment:
CI fails because it tries to run this test on all the OTP versions. I think what you can do is to:

1. create a function that checks that ed25519 is supported
    ed25519_supported() ->
      {ok, PublicKeyAppVersionString} = application:get_key(public_key, vsn),
      PublicKeyAppVersionList = lists:map(fun erlang:list_to_integer/1, string:tokens(PublicKeyAppVersionString, ".")),
      PublicKeyAppVersionList < [1, 11, 2].
2. Use this function in `dkim_get_algorithm_digest`
3. Use this function in this test and if `ed25519_supported()` returns `false`, return an empty list from `dkim_sign_ed25519_test_()`
4. Might be good to add `24.1` to https://github.com/gen-smtp/gen_smtp/blob/master/.github/workflows/ci.yml so we actually run this test

Reply:
Test will also fail as it isn't completed. Still a copy of RSA test.
Thanks for guiding. I will do the changes in my next session.
Can I upgrade OTP 24.0 to 24.1 in CI? Or is there anything that both versions are needed?

Reply:
I think replacing 24.0 with 24.1 should be fine